.github/workflows/build-container-image.yaml

```diff
@@ -0,0 +1,40 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow
+
+name: Create and publish a Docker image
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
```

Comment:
Why is write access needed if we don't push anything?

Reply:
This would also prevent to accidentally set `push: true` (see below).

Reply:
I removed the permission with 29d76c62c697d41dd2bbc44b9c4bf458a78d7481